### PR TITLE
Tech stacks

### PR DESCRIPTION
**Pull request by dharmelolar**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by Dusch4593** [comment]
Hey @dharmelolar, thank you again for taking on this entry! 🚀 I'll conduct the first review for this PR and tag you here once I'm finished. 😄

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (Dusch4593) [inline content/general/concepts/tech-stacks/tech-stacks.md:31]:
```suggestion

## Common Stacks

### LAMP
```

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (Dusch4593) [inline content/general/concepts/tech-stacks/tech-stacks.md:28]: Any chance we could move/incorporate this information into the existing general [back-end entry](https://www.codecademy.com/resources/docs/general/back-end)?

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (Dusch4593) [inline content/general/concepts/tech-stacks/tech-stacks.md:21]:
```suggestion
The[ back-end](https://www.codecademy.com/resources/docs/general/back-end), or server-side, refers to parts of an application's code that allow it to operate and handle new and existing data. This cannot usually be accessed by a user. The back-end of a computer system is where the majority of data and operating syntax are stored and accessed.

The back-end tech stack is usually comprised of programming languages, [frameworks](https://www.codecademy.com/resources/docs/general/framework), [databases](https://www.codecademy.com/resources/docs/general/database), and [servers](https://www.codecademy.com/resources/docs/general/server).
```

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (Dusch4593) [inline content/general/concepts/tech-stacks/tech-stacks.md:32]:
```suggestion
The LAMP stack is a popular open-source tech stack that is used to create web apps. It is one of the oldest tech stacks and provides a collection of software for delivering high-performance web applications. This stack comprises the following technologies:
```

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (Dusch4593) [inline content/general/concepts/tech-stacks/tech-stacks.md:37]:
```suggestion
- Linux: An open-source operating system based on the Linux kernel
- Apache: A free and open-source cross-platform web server.
- MySQL: A relational database management system for storing and manipulating data.
- PHP: A server-side scripting programming language used for creating dynamic web applications.
```

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (Dusch4593) [reply content/general/concepts/tech-stacks/tech-stacks.md:39]: For lines # 39 through 63, I think we're repeating some content when describing MongoDB, Express, and Node.js. Could we consolidate the three "ME_N" sections into one section entitled "MERN And Alternative Stacks" that has just one unordered list describing the different softwares used in the stack? Then when talking about MEAN and MEVN we could simply mention that Angular and Vue are the front-end technologies used, respectively.

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (Dusch4593) [inline content/general/concepts/tech-stacks/tech-stacks.md:45]:
```suggestion
- MongoDB is a document database used to build highly available and scalable internet applications.
- Express(.js) is a back-end framework built on Node.js and it provides features for developing mobile and web applications.
- [React(.js)](https://www.codecademy.com/resources/docs/react) is a front-end library developed by Meta for building user interface.
- Node(.js) is a runtime environment that runs JavaScript on the server-side.
```

We do have existing entries for Express.js and Node.js but I'm not going to suggest linking them at this time because they are in the process of the being moved and having their URLs changed.

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@dharmelolar Finished with my review and left you some comments/suggestions. Let me know if you have any questions or have finished implementing the feedback. 😄

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (dharmelolar) [reply content/general/concepts/tech-stacks/tech-stacks.md:64]: Do I take out this part from my entry?

**Comment by dharmelolar** [comment]
@Dusch4593 I've implemented the feedback and suggestions. Thank you!

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (Dusch4593) [inline content/general/concepts/tech-stacks/tech-stacks.md:23]:
```suggestion
### Back-End Tech Stack

```

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (Dusch4593) [inline content/general/concepts/tech-stacks/tech-stacks.md:15]:
```suggestion

A **technology (or "tech") stack** refers to a set of softwares that work together to build digital products or solutions such as websites and mobile applications. Essential for building maintainable and scalable software, tech stacks are also referred to as technology infrastructures or solution stacks.
```

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (Dusch4593) [inline content/general/concepts/tech-stacks/tech-stacks.md:43]:
```suggestion
- MongoDB: A document database used to build highly available and scalable internet applications.
```

**Review by Dusch4593 — APPROVED** [review]
@dharmelolar LGTM! I left a few small suggestions (mostly to do with spacing) but we are good for the next review. 😄

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (SSwiniarski) [inline content/general/concepts/tech-stacks/tech-stacks.md:48]: Again, it would be nice to mention front-end vs. back-end here.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@dharmelolar, I did a second review and only had a couple of comments.

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (dharmelolar) [reply content/general/concepts/tech-stacks/tech-stacks.md:48]: I think I've explained this from my last commit. Express is the backend framework while react/vue/angular are the frontend frameworks.

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (dharmelolar) [reply content/general/concepts/tech-stacks/tech-stacks.md:39]: The lamp stack is a backend stack that's why I couldn't mention which is frontend

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (SSwiniarski) [reply content/general/concepts/tech-stacks/tech-stacks.md:48]: Am I missing where it says react/vue/angular are the frontend frameworks?

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (SSwiniarski) [reply content/general/concepts/tech-stacks/tech-stacks.md:39]: Then maybe mention this in the introductory paragraph.

**Comment by dharmelolar** [comment]
@SSwiniarski you can check now

**Review by SSwiniarski — APPROVED** [review]
@dharmelolar, looks good now.

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (HishamT) [inline content/general/concepts/tech-stacks/tech-stacks.md:34]: "The LAMP stack is a popular back-end tech stack ~that is~ used to create web apps. This stack ~comprises~ contains the following technologies:"

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (HishamT) [inline content/general/concepts/tech-stacks/tech-stacks.md:43]: ~MERN is used to describe a specific set of JavaScript-based technologies that are used in the web application development process. It is made of technologies such as MongoDB, Express, React, and Node.~

"MERN is a set of JavaScript-based technologies used in the web application development process. MERN is the collection of the following technologies:
- MongoDB: A document database used to build highly available and scalable internet applications.
- Express (.js): A ~backend~ back-end framework built on Node.js ~and it~ providing features for developing mobile and web applications.
- [React(.js)](https://www.codecademy.com/resources/docs/react) / Vue.js / Angular.js : These are ~frontend~ front-end javascript frameworks for building user interfaces and single-page applications.
- Node(.js) is a runtime environment that runs JavaScript on the server-side." 


"The only difference ~is~ being Vue.js or Angular.js is used in place of React.js as the front-end framework.

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (HishamT) [inline content/general/concepts/tech-stacks/tech-stacks.md:44]: The following definitions should be made above as shown in a previous comment.

File: content/general/concepts/tech-stacks/tech-stacks.md
Comment (HishamT) [inline content/general/concepts/tech-stacks/tech-stacks.md:38]: "MySQL: A relational database management system for storing and manipulating data on the ~backend~ back-end."

**Review by HishamT — CHANGES_REQUESTED** [review]
Looks pretty good, but I would like some changes done.

**Comment by dharmelolar** [comment]
Made the changes @HishamT

**Review by HishamT — APPROVED** [review]
LGTM!

**Comment by Dusch4593** [comment]
@dharmelolar Looks like we have enough approvals for merge! 🎉 I made a few last-minute edits in 9ba1d96ea8d7425347a6e30a0b60b5792ac79a89 that I wanted to share with you:

- Changed folder name, file name, and `Title` to singular "Tech Stack" since most other concepts under General are singular rather than plural.
- Implemented a few lingering pieces of feedback from a previous review.
- Fixed hyperlink spacing on line # 26.
- Added period to end of list item on line # 36.
- Consolidated text from lines # 43 and 50 to one line.
- Changed "front-end framework(s)" to "front-end technology/ies" since React technically isn't a bona fide framework, but a library.
- On line # 48, I added the colon `:` next to "Node(.js)".

cc: @SSwiniarski @HishamT

**Comment by Dusch4593** [comment]
Congratulations, @dharmelolar, your contribution should now be live! 🎉 🙌🏻 🎊 🚀

Live link here: https://www.codecademy.com/resources/docs/general/tech-stack

I noticed that your Codecademy profile showed a single letter "D":

Screen Shot 2022-09-29 at 2 56 33 PM

Was this the profile image you'd intended?

**Comment by dharmelolar** [comment]
Thank you so much for your help and cheers to more contributions🎉

My codecademy account is linked to my Google account that’s why the image is like that.

**Pull request merged**
